Source/Scene/Cesium3DTileGroundPrimitive.js

```diff
@@ -0,0 +1,875 @@
+            frontOperation : {
+                fail : StencilOperation.KEEP,
+                zFail : StencilOperation.KEEP,
+                zPass : StencilOperation.DECREMENT_WRAP
+            },
+            backFunction : StencilFunction.NOT_EQUAL,
+            backOperation : {
+                fail : StencilOperation.KEEP,
+                zFail : StencilOperation.KEEP,
+                zPass : StencilOperation.DECREMENT_WRAP
+            },
+            reference : 0,
+            mask : ~0
+        },
+        depthTest : {
+            enabled : false
+        },
+        depthMask : false
+    };
+
+    function createRenderStates(primitive) {
+        if (defined(primitive._rsStencilPreloadPass)) {
+            return;
+        }
+
+        primitive._rsStencilPreloadPass = RenderState.fromCache(stencilPreloadRenderState);
+        primitive._rsStencilDepthPass = RenderState.fromCache(stencilDepthRenderState);
+        primitive._rsColorPass = RenderState.fromCache(colorRenderState);
+        primitive._rsPickPass = RenderState.fromCache(pickRenderState);
+    }
+
+    var modifiedModelViewScratch = new Matrix4();
+    var rtcScratch = new Cartesian3();
+
+    function createUniformMap(primitive, context) {
+        if (defined(primitive._uniformMap)) {
+            return;
+        }
+
+        primitive._uniformMap = {
+            u_modifiedModelViewProjection : function() {
+                var viewMatrix = context.uniformState.view;
+                var projectionMatrix = context.uniformState.projection;
+                Matrix4.clone(viewMatrix, modifiedModelViewScratch);
+                Matrix4.multiplyByPoint(modifiedModelViewScratch, primitive._center, rtcScratch);
+                Matrix4.setTranslation(modifiedModelViewScratch, rtcScratch, modifiedModelViewScratch);
+                Matrix4.multiply(projectionMatrix, modifiedModelViewScratch, modifiedModelViewScratch);
+                return modifiedModelViewScratch;
+            }
+        };
+    }
+
+    function copyIndices(indices, newIndices, currentOffset, offsets, counts, batchedIds) {
+        var sizeInBytes = indices.constructor.BYTES_PER_ELEMENT;
+
+        var batchedIdsLength = batchedIds.length;
+        for (var j = 0; j < batchedIdsLength; ++j) {
+            var batchedId = batchedIds[j];
+            var offset = offsets[batchedId];
+            var count = counts[batchedId];
+
+            var subarray = new indices.constructor(indices.buffer, sizeInBytes * offset, count);
+            newIndices.set(subarray, currentOffset);
+
+            offsets[batchedId] = currentOffset;
+            currentOffset += count;
+        }
+
+        return currentOffset;
+    }
+
+    function rebatchCommands(primitive) {
+        var batchedIndices = primitive._batchedIndices;
+        var length = batchedIndices.length;
+
+        var needToRebatch = false;
+        var colorCounts = {};
+
+        for (var i = 0; i < length; ++i) {
```

Comment:
What is `length`?  Number of materials/colors in the tile?  This could burn a good bit of CPU in the steady state when nothing changed.  Is it possible to protect this with a dirty flag?